Sources/Sentry/SentryFileManager.m

```diff
@@ -202,20 +202,13 @@ - (NSURL *)getSentryPathAsURL
 
 - (nullable NSString *)storeEnvelope:(SentryEnvelope *)envelope
 {
-    NSData *envelopeData = [SentrySerialization dataWithEnvelope:envelope];
-
-    if (envelopeData == nil) {
-        SENTRY_LOG_ERROR(@"Serialization of envelope failed. Can't store envelope.");
-        return nil;
-    }
-
     @synchronized(self) {
         NSString *path =
             [self.envelopesPath stringByAppendingPathComponent:[self uniqueAscendingJsonName]];
         SENTRY_LOG_DEBUG(@"Writing envelope to path: %@", path);
 
-        if (![self writeData:envelopeData toPath:path]) {
-            SENTRY_LOG_WARN(@"Failed to store envelope.");
+        if (![SentrySerialization writeEnvelope:envelope toPath:path]) {
```

Comment:
`m`: In addition to the other comment, I would propose the following (if reasonable):

`SentryEnvelopePersistentStore` offers the `storeEnvelope` method which uses the `SentrySerialization` to convert the data and passes it to `SentryFileManager` to write it (see `SentryScopeContextPersistentStore` for reference).

That way we have a lower-level class to access the disk, a helper class to convert data structures to binary data and a wrapper-class with a simpler interface.

Reply:
I think it makes more sense to do this on a follow up, not in this PR.
Mainly because this will need changes in many places and might make it harder to review